Sources/ArcGISToolkit/Components/Offline/PreplannedListItemView.swift

```diff
@@ -27,6 +27,22 @@ struct PreplannedListItemView: View {
     /// A Boolean value indicating whether the metadata view is presented.
     @State private var metadataViewIsPresented = false
     
+    /// The download state of the preplanned map model.
+    private enum DownloadState {
+        case notDownloaded, downloading, downloaded
+        
+        init(_ state: PreplannedMapModel.Status) {
+            self = switch state {
+            case .downloaded: .downloaded
+            case .downloading: .downloading
+            default: .notDownloaded
+            }
+        }
```

Comment:
I just showed this here for simplicity, but normally we would keep an enum/struct declaration clean of convenience members and instead add them in private extensions towards the bottom of the file. Consider doing that here to keep the view itself as simple and readable as possible.

Reply:
You misunderstood what I meant. I was referring to this initializer (the one that I selected) only and not the entire enum. The enum itself should be where it was, but this initializer makes sense in an extension:

```swift
private extension PreplannedListItemView.DownloadState {
    init(_ state: PreplannedMapModel.Status) {
        ...
    }
}
```